src/build/env.py

```diff
@@ -1,312 +1,313 @@
-                '-r',
-                os.path.abspath(req_file.name),
-            ]
-            _subprocess(cmd)
-        finally:
-            os.unlink(req_file.name)
+    def create(self) -> None:
+        if self._env_created:
+            return
 
+        if _should_use_virtualenv():
+            self._log('Creating isolated environment (virtualenv)...')
+            self._python_executable, self._scripts_dir = _create_isolated_env_virtualenv(self._isolated_env_manager.path)
+        else:
+            self._log('Creating isolated environment (venv)...')
+            self._python_executable, self._scripts_dir = _create_isolated_env_venv(self._isolated_env_manager.path)
+            self._patch_up_venv()
 
-def _create_isolated_env_virtualenv(path: str) -> Tuple[str, str]:
-    """
-    On Python 2 we use the virtualenv package to provision a virtual environment.
+        self._env_created = True
 
-    :param path: The path where to create the isolated build environment
-    :return: The Python executable and script folder
-    """
-    cmd = [str(path), '--no-setuptools', '--no-wheel', '--activators', '']
-    result = virtualenv.cli_run(cmd, setup_logging=False)
-    executable = str(result.creator.exe)
-    script_dir = str(result.creator.script_dir)
-    return executable, script_dir
+    def _patch_up_venv(self) -> None:
+        self._log('Updating pip...')
+        self._run([self.python_executable, '-m', 'pip', 'install', f'pip>={_get_min_pip_version()}'])
+        self._log('Uninstalling setuptools...')
+        self._run([self.python_executable, '-m', 'pip', 'uninstall', '-y', 'setuptools'])
 
+    def install_packages(self, requirements: Iterable[str]) -> None:
+        req_list = sorted(requirements)
+        if not req_list:
+            return
 
-@functools.lru_cache(maxsize=None)
-def _fs_supports_symlink() -> bool:
-    """Return True if symlinks are supported"""
-    # Using definition used by venv.main()
-    if os.name != 'nt':
-        return True
+        # pip does not honour environment markers in command line arguments;
+        # it does for requirements from a file.
+        with tempfile.NamedTemporaryFile(
+            'w',
+            prefix='build-reqs-',
+            suffix='.txt',
+            # On Windows the temp file can't be opened by pip while it is kept open by build
+            # so we defer deleting it.
+            delete=False,
+        ) as req_file:
+            req_file.write(os.linesep.join(req_list))
 
-    # Windows may support symlinks (setting in Windows 10)
-    with tempfile.NamedTemporaryFile(prefix='build-symlink-') as tmp_file:
-        dest = f'{tmp_file}-b'
         try:
-            os.symlink(tmp_file.name, dest)
-            os.unlink(dest)
-            return True
-        except (OSError, NotImplementedError, AttributeError):
-            return False
+            self._log(f'Installing build dependencies... ({", ".join(req_list)})')
+            self._run(
```

Comment:
This would read better if we'd declare first the command, and fit it in two lines. the comment can go after definining the cmd variable.

Reply:
The `--pep-517` comment? But that's specific to that option?

Reply:
I know but it's kind obvious what you're talking about it, doesn't need IMHO to be right beside it.